src/conn/conn_api.c

```diff
@@ -1149,6 +1150,13 @@ __conn_reconfigure(WT_CONNECTION *wt_conn, const char *config)
 	__wt_free(session, conn->cfg);
 	conn->cfg = p;
 
+	/*
+	 * Fourth, run any diagnostic commands that were included. Do that last,
+	 * reconfiguration might include other configuration changes to support
+	 * a command.
+	 */
+	WT_ERR(__conn_verbose_cmd(session, config));
```

Comment:
> These APIs aren't likely to be called frequently: I think admins are more likely to commonly reconfigure more eviction worker threads, for example.

I generally do not agree with this statement.  I doubt any admins or mongodb support people have reconfigured eviction worker threads much other than an explicit direction to do so from someone on our team in some HELP ticket.  Given the number of HELP tickets that discuss eviction, and once this cat is out of the bag, I can see this being used more and more frequently.

Basically I agree with @agorrod that short-circuiting and avoiding thread disruption on verbose-only is a good idea.  I think it will be used a lot more often than you're expecting.